python/paddle/fluid/backward.py

```diff
@@ -1215,12 +1298,35 @@ def _as_list(x):
     return list(x) if isinstance(x, collections.Sequence) else [x]
 
 
+def _get_output_names(block, outputs):
+
+    current_output_names = set([out.name for out in outputs])
+
+    if outputs:
+        current_block = outputs[0].block
+        while current_block.idx != block.idx:
```

Comment:
I'm not quite understand these lines of codes. Please correct me if I am wrong about following comments:

1. link_output_names = set() at every beginning of while loop and current_output_names = link_output_names at every end of while loop. So the assignment to current_output_names actually takes effect at only the last iteration. Is there something wrong you write the while loop here?

2. The  while loop code can be wrong if caller inputs with "outputs[0].block" which is an ancestor of "block". Make an assert and throw some exception message for checking that case?

Reply:
> 1. link_output_names = set() at every beginning of while loop and current_output_names = link_output_names at every end of while loop. So the assignment to current_output_names actually takes effect at only the last iteration. Is there something wrong you write the while loop here?

(I modified variable names to make it easier to understand the while loop)


> 2. The while loop code can be wrong if caller inputs with "outputs[0].block" which is an ancestor of "block". Make an assert and throw some exception message for checking that case?

done.